x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/crud/rules_management_client.ts

```diff
@@ -0,0 +1,379 @@
+  existingRule: RuleAlertType;
+  ruleUpdate: RuleUpdateProps;
+}
+
+export interface _PatchRuleProps {
+  existingRule: RuleAlertType;
+  nextParams: PatchRuleRequestBody;
+}
+
+interface CreateCustomRuleProps {
+  params: RuleCreateProps;
+}
+
+interface CreatePrebuiltRuleProps {
+  ruleAsset: PrebuiltRuleAsset;
+}
+
+type UpdateRuleProps = _UpdateRuleProps;
+
+type PatchRuleProps = _PatchRuleProps;
+
+interface DeleteRuleProps {
+  ruleId: RuleObjectId;
+}
+
+interface UpgradePrebuiltRuleProps {
+  ruleAsset: PrebuiltRuleAsset;
+}
+
+interface ImportRuleOptions {
+  allowMissingConnectorSecrets?: boolean;
+}
+
+interface ImportRuleProps {
+  ruleToImport: RuleToImport;
+  overwriteRules?: boolean;
+  options: ImportRuleOptions;
+}
+
+interface ImportNewRuleProps {
+  ruleToImport: RuleToImport;
+  options: ImportRuleOptions;
+}
+
+interface ImportExistingRuleProps {
+  ruleToImport: RuleToImport;
+  existingRule: RuleAlertType;
+}
+
+export interface IRulesManagementClient {
+  createCustomRule: (createCustomRulePayload: CreateCustomRuleProps) => Promise<RuleAlertType>;
+  createPrebuiltRule: (
+    createPrebuiltRulePayload: CreatePrebuiltRuleProps
+  ) => Promise<RuleAlertType>;
+  updateRule: (updateRulePayload: UpdateRuleProps) => Promise<RuleAlertType>;
+  patchRule: (patchRulePayload: PatchRuleProps) => Promise<RuleAlertType>;
+  deleteRule: (deleteRulePayload: DeleteRuleProps) => Promise<void>;
+  upgradePrebuiltRule: (
+    upgradePrebuiltRulePayload: UpgradePrebuiltRuleProps
+  ) => Promise<RuleAlertType>;
+  importRule: (importRulePayload: ImportRuleProps) => Promise<RuleAlertType>;
+}
+
+export const createRulesManagementClient = (rulesClient: RulesClient) => {
+  const client = {
+    createCustomRule: async (
+      createCustomRulePayload: CreateCustomRuleProps
+    ): Promise<RuleAlertType> => {
+      return createCustomRule(rulesClient, createCustomRulePayload);
+    },
+
+    createPrebuiltRule: async (
+      createPrebuiltRulePayload: CreatePrebuiltRuleProps
+    ): Promise<RuleAlertType> => {
+      return createPrebuiltRule(rulesClient, createPrebuiltRulePayload);
+    },
+
+    updateRule: async (updateRulePayload: UpdateRuleProps): Promise<RuleAlertType> => {
+      return updateRule(rulesClient, updateRulePayload);
+    },
+
+    patchRule: async (patchRulePayload: PatchRuleProps): Promise<RuleAlertType> => {
+      return patchRule(rulesClient, patchRulePayload);
+    },
+
+    deleteRule: async (deleteRulePayload: DeleteRuleProps): Promise<void> => {
+      return deleteRule(rulesClient, deleteRulePayload);
+    },
+
+    upgradePrebuiltRule: async (
+      upgradePrebuiltRulePayload: UpgradePrebuiltRuleProps
+    ): Promise<RuleAlertType> => {
+      return upgradePrebuiltRule(rulesClient, upgradePrebuiltRulePayload);
+    },
+
+    importRule: async (importRulePayload: ImportRuleProps): Promise<RuleAlertType> => {
+      return importRule(rulesClient, importRulePayload);
+    },
```

Comment:
All these methods return `RuleAlertType`, a lower-level type used by the Alerting framework. Do we actually use this type, or can we replace it with `RuleResponse`?

Reply:
`RuleAlertType` is correct in this case; at least, the way it is implemented now, this new client responds with the rules in the internal data schema (camelCase fields, params object, etc.)

In our endpoints, we later transform this into `RuleResponse` with the `transformValidate` function, just before returning them.

I get your point, and this should probably happen within the `RulesManagementClient`, I'm just a little hesitant to do it within this PR instead of in a follow up.